app/src/main/java/com/example/iou/activities/SignupActivity.java

```diff
@@ -0,0 +1,87 @@
+package com.example.iou.activities;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.iou.MainActivity2;
+import com.example.iou.R;
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class SignupActivity extends AppCompatActivity {
+
+    public static final String TAG = "SignupActivity";
+    private EditText etUsernameSignup;
+    private EditText etPasswordSignup;
+    private Button btnSignupSignup;
```

Comment:
You should mark these variables as `final` to indicate that they won't be re-assigned. (& ditto for the class variables in your other files!)

Reply:
Would I need to make a constructor for these variables then? when I'm putting final in front of the variables Android Studio is giving me an error.

Reply:
Ah that's right, you can't mark these as `final` here because they need to be assigned in the `onCreate()` function. My bad, no need to make changes here then!